frontend/lib/src/dataframes/arrowFormatUtils.ts

```diff
@@ -536,45 +542,57 @@ function formatInterval(x: StructRow, field?: Field): string {
  * @returns The formatted cell value.
  */
 export function format(x: DataType, type: ArrowType): string {
-  if (isNullOrUndefined(x)) {
-    return ""
-  }
+  try {
+    if (isNullOrUndefined(x)) {
+      return ""
+    }
 
-  const isDate = x instanceof Date || Number.isFinite(x)
-  if (isDate && isDateType(type)) {
-    return formatDate(x as Date | number)
-  }
+    if (isStringType(type)) {
+      return String(x)
+    }
 
-  if (typeof x === "bigint" && isTimeType(type)) {
-    return formatTime(Number(x), type.arrowField)
-  }
+    const isDate = x instanceof Date || Number.isFinite(x)
+    if (isDate && isDateType(type)) {
+      return formatDate(x as Date | number)
+    }
 
-  if (isDate && isDatetimeType(type)) {
-    return formatDatetime(x as Date | number, type.arrowField)
-  }
+    if (typeof x === "bigint" && isTimeType(type)) {
+      return formatTime(Number(x), type.arrowField)
+    }
 
-  if (isPeriodType(type)) {
-    return formatPeriod(x as bigint, type.arrowField)
-  }
+    if (isDate && isDatetimeType(type)) {
+      return formatDatetime(x as Date | number, type.arrowField)
+    }
 
-  if (isIntervalType(type)) {
-    return formatInterval(x as StructRow, type.arrowField)
-  }
+    if (isPeriodType(type)) {
+      return formatPeriod(x as bigint, type.arrowField)
+    }
 
-  if (isDurationType(type)) {
-    return formatDuration(x as number | bigint, type.arrowField)
-  }
+    if (isIntervalType(type)) {
+      return formatInterval(x as StructRow, type.arrowField)
+    }
 
-  if (isDecimalType(type)) {
-    return formatDecimal(x as Uint32Array, type.arrowField)
-  }
+    if (isDurationType(type)) {
+      return formatDuration(x as number | bigint, type.arrowField)
+    }
 
-  if (isFloatType(type) && Number.isFinite(x)) {
-    return formatFloat(x as number)
-  }
+    if (isDecimalType(type)) {
+      return formatDecimal(x as Uint32Array, type.arrowField)
+    }
 
-  if (isObjectType(type) || isListType(type)) {
-    return formatObject(x, type.arrowField)
+    if (isFloatType(type) && Number.isFinite(x)) {
+      return formatFloat(x as number)
+    }
+
+    if (isObjectType(type) || isListType(type)) {
+      return formatObject(x, type.arrowField)
```

Comment:
This part is the same logic as before, the main change here is to add some more safeguards / fallbacks against eventual errors during data handling. The formatting here should always return a value and never throw an error.